synchronized/lib/synchronized.dart

```diff
@@ -48,8 +48,8 @@ abstract class Lock {
   /// Executes [computation] when lock is available.
   ///
   /// Only one asynchronous block can run while the lock is retained.
-  Future<T> synchronized<T>(FutureOr<T> Function() computation,
-      {Duration timeout});
+  Future<T?> synchronized<T>(FutureOr<T?> Function()? computation,
```

Comment:
What is the use case for passing a `null` computation here? Does that make sense or was maybe the migration tool going off the rails a bit (it can get confused by for instance tests passing null)?

Reply:
There is actually a test case for that. Passing null will wait for the previous task to terminate. It is similar to a () {} function. Yes I will likely remove the null case in the future for computation (for now i did not want to change the behavior and unit test)

Reply:
Ya ok that makes sense - if you did change the api later I would suggest a separate api entirely for this behavior - that would allow you to remove the `null as T` cast from my previous comment.

Reply:
>  if you did change the api later I would suggest a separate api entirely for this behavior

Yes I guess not supporting null will be a breaking change so would require a major version bump. I'm not sure if you suggest an api to wait for any pending tasks (alias to synchronized(() {}) and keeping null support or breaking null support?